parser.go

```diff
@@ -8,21 +8,113 @@ import (
 	"unicode"
 )
 
+type parserLang struct {
+	lang            Language
+	prevWhitespace  uint64
+	prevMode        uint
+	prevIsIdentRune func(ch rune, i int) bool
+}
+
+type parserLangStack []parserLang
+
+func (pls *parserLangStack) push(pl parserLang) {
+	*pls = append(*pls, pl)
+}
+
+func (pls parserLangStack) peek() (parserLang, bool) {
+	if len(pls) == 0 {
+		return parserLang{}, false
+	}
+
+	return pls[len(pls)-1], true
+}
+
+func (pls *parserLangStack) pop() (parserLang, bool) {
+	pl, ok := pls.peek()
+	if !ok {
+		return parserLang{}, false
+	}
+
+	*pls = (*pls)[:len(*pls)-1]
+	return pl, true
+}
+
 //Parser parses expressions in a Language into an Evaluable
 type Parser struct {
-	scanner scanner.Scanner
-	Language
```

Comment:
This is embeded type is public. I have a a fast and elegant solution in my mind. So I would fix this myself.

Reply:
Ah, sorry about that! Let me know if you'd like me to target this PR to a different branch so you can handle that. I can also give you write access to our repo if it'd be more convenient for you.

Reply:
No problem.

I can edit the target myself.